arrow-flight/src/lib.rs

```diff
@@ -459,6 +511,51 @@ impl FlightInfo {
         let msg = IpcMessage(self.schema);
         msg.try_into()
     }
+
+    /// Specify the schema for the response.
+    ///
+    /// Note this takes the arrow [`Schema`] (not the IPC schema) and
+    /// encodes it using the default IPC options.
+    ///
+    /// Returns an error if `schema` can not be encoded into IPC form.
+    pub fn with_schema(mut self, schema: &Schema) -> ArrowResult<Self> {
```

Comment:
Just minor, but personally, I like to make the the fact that its fallible explicit in the name.

```suggestion
    pub fn try_with_schema(mut self, schema: &Schema) -> ArrowResult<Self> {
```

Reply:
in a6c973f13